src/main/java/htsjdk/variant/vcf/VCFPathReader.java

```diff
@@ -0,0 +1,183 @@
+package htsjdk.variant.vcf;
+
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMSequenceDictionary;
+import htsjdk.samtools.util.CloseableIterator;
+import htsjdk.samtools.util.CloserUtil;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.IntervalList;
+import htsjdk.tribble.AbstractFeatureReader;
+import htsjdk.tribble.FeatureCodec;
+import htsjdk.tribble.FeatureReader;
+import htsjdk.tribble.TribbleException;
+import htsjdk.variant.bcf2.BCF2Codec;
+import htsjdk.variant.variantcontext.VariantContext;
+
+import java.io.Closeable;
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+
+/**
+ * Simplified interface for reading from VCF/BCF files.
+ */
+public class VCFPathReader implements Closeable, Iterable<VariantContext> {
+
+    private final FeatureReader<VariantContext> reader;
+
+    /**
+     * Returns true if the given path appears to be a BCF file.
+     */
+    public static boolean isBCF(final Path path) {
+        return path.toUri().getRawPath().endsWith(".bcf");
```

Comment:
Why use `.toUri()` here and below? In this case you could write `path.toString().endsWIth(".bcf")`

Reply:
because I wanted to remove the possible junk after a ? in a URL.

Reply:
In what case does a `Path` have a query string? A `Path` is not a URI and a query string would be invalid, as far as I know.

Reply:
I think that path could have a query string...but I could be wrong.